src/lib/environment.js

```diff
@@ -0,0 +1,29 @@
+let state = {
+  isArchiveMode: false,
+  pathToProject: '.',
+};
+
+const initialize = (pathToProject, isArchiveMode, archiveDestination, extraArgs, apiUrl) => {
+  if (state.initialized) {
+    throw Error('environment is already initialized');
+  }
+
+  state = {
+    apiUrl,
+    archiveDestination,
+    extraArgs,
+    initialized: true,
+    isArchiveMode,
+    pathToProject,
+  };
+};
+
+module.exports = {
+  initialize,
+
+  get pathToProject() { return state.pathToProject; },
+  get isArchiveMode() { return state.isArchiveMode; },
+  get archiveDestination() { return state.archiveDestination; },
+  get extraArgs() { return state.extraArgs; },
+  get apiUrl() { return state.apiUrl; },
+};
```

Comment:
I personally don't find getters to be strange at all here. They seem like a solid option for a read-only interface to an existing value. If we should avoid the use of getters in code, maybe we can add that into our style guide?

Reply:
Traditionally getters are used to classes to access instance properties. These are static constants (effectively) so there is only one instance. The functional approach also guarantees readonly access so there's no practical difference between your suggestion and mine, it's just syntactic preference. I'm trying to avoid introducing a new style for the same pattern we use elsewhere.

I'm not ready to mandate anything in the style guide, and I'm happy to disagree and commit as this is just a personal preference. I'll leave it for you to decide.